### ISSUE TITLE
Add support for Composer v2

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#2276]
## Feature Description

Composer v2 was [just released](https://blog.packagist.com/composer-2-0-is-now-available/) and the announcement includes a recommendation for everyone to adopt v2 as soon as possible as v1 will soon be EOL.

This is mostly a seamless upgrade for us as it is more/less no different from a project perspective. The main things to consider are:
* **support/updates** - As mentioned above, v1 will be end of life soon so we should move to using v2 barring any major blockers
* **PHP version support** - v2 currently supports PHP 5.3+ but this will be dropped soon to only support non-EOL versions of PHP. More on this below
* **Dependencies which depend on v1 of the plugin API** - This doesn't affect a non-development build for us, but specifically the current version of PHP Scoper has a dependency that install isn't compatible with Cv2

Regarding future PHP support
> As I mentioned above, Composer 2.0 supports PHP 5.3+, which is at this point very outdated and makes the code quite hard to maintain in places. We went through the effort to make sure every Composer user can upgrade to Composer 2, but the plan is to drop support for EOL PHP versions in a future minor release.
>
> Composer 2.1 might still come with PHP 5.3 support, depending on the timeline and which features end up being included, but then at the latest by Composer 2.2 we will drop support for everything older than PHP 7.1.3. According to our stats this allows over 90% of the Composer users to use the latest version, and **for the others who are stuck on outdated PHP versions we will keep providing critical bug and security fixes in the 2.0.x or 2.1.x range**.

Even the (future) PHP version limitation is not of much concern for us as we run Composer in a docker container in CI anyways to meet the minimum version of PHP required by PHP Scoper which requires PHP 7.2.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* `composer install` should work with Composer v1 & v2
* Docker-based `composer install` in CI should be updated to use the `composer:2.0` image

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
Closing this as no longer necessary after the follow-up PR to #2270
